ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -837,126 +794,100 @@ def set_waypoints_velocity(self):
-                    recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
-                                               self.lookahead_wps,
-                                               dist_to_tl - (self.dyn_tl_buffer - 1.0))
-
-            elif dist_to_tl < self.dyn_creep_zone + self.dyn_tl_buffer:
-                if self.waypoints[self.final_waypoints_start_ptr].get_v() <=\
-                        self.handoff_velocity:
-                    self.set_creep(self.final_waypoints_start_ptr, self.lookahead_wps)
-                else:
-                    if self.state != 'slowdown' and self.state != 'stopped':
-                        rospy.logwarn("now start to slowdown")
-                    recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
-                                               self.lookahead_wps,
-                                               dist_to_tl - (self.dyn_tl_buffer - 1.0))
-                    self.stop_target = self.next_tl_wp
             else:
-                # not sure when this might trigger but not match next if statement - then 
-                # reduce car velocity to 0.0 stoping before lights
-                # if self.state != 'slowdown' and self.state != 'stopped':
-                #    rospy.logwarn("set car state to slowdown at ptr = {}".format(self.final_waypoints_start_ptr))
-
-                if dist_to_tl - self.dyn_tl_buffer > self.min_stop_distance:
-                    if self.state != 'slowdown':
-                        rospy.logwarn("Within stopping distance of TL - Start to slowdown!")
-                        self.stop_target = self.next_tl_wp
-                    else:
-                        rospy.logwarn("we should not get here! {}, {}".format(self.stop_target, self.next_tl_wp))
-                    recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
-                                               self.lookahead_wps,
-                                               dist_to_tl - (self.dyn_tl_buffer - 1.0))
+                recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
+                                           self.lookahead_wps,
+                                           dist_to_tl - (self.dyn_tl_buffer - 1.0))
+
+        # switch to slowdown if possible!
+        elif (self.state == 'speedup' or self.state == 'maintainspeed') and\
+               dist_to_tl - self.dyn_tl_buffer < self.stopping_distance :
+            if dist_to_tl - self.dyn_tl_buffer > self.min_stop_distance:
+                if self.state != 'slowdown':
+                    rospy.logwarn("Within stopping distance of TL - Start to slowdown!")
+                self.stop_target = self.next_tl_wp
+                recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
+                            self.lookahead_wps,
+                            dist_to_tl - (self.dyn_tl_buffer - 1.5))
```

Comment:
I'm getting stuck in slowdown state for small waypoint bump (+40) slowdowns, maybe because of the reduced default accel rate?  Looks like you increased the target stopping point a little here, but it's still `- 1.0` on line 833.  Maybe this should be a parameter?  I am able to get stopping state to trigger each time if I move it all the way with no buffer (if we set this parameter all the way to = self.dyn_tl_buffer) but not sure if that's the best way.

Maybe we could just boost the stopping zone threshold to judge stopped state on line 826 like:
`if dist_to_tl < (self.dyn_tl_buffer * 1.1) and` to give it some more margin instead of tweaking the slowdown profile accel rates or moving the stopping point around?

Reply:
Boosting the stopping zone threshold also required the same boost on line 819 to prevent a quick creep after stopping to get into the zone, but then the speed at stopping got pretty high since it happens farther away.  Maybe tuning the target stopping point offset as a parameter is better.